cloudify_cli/commands/install.py

```diff
@@ -0,0 +1,180 @@
+            # the archive file without the extension
+            if archive_location_type == 'path':
+                filename, ext = os.path.splitext(
+                    os.path.basename(archive_location))
+                blueprint_id = filename
+
+            # if the archive is a url, assign blueprint_id name of the file
+            # that the url leads to, without the extension.
+            # e.g. http://example.com/path/archive.zip?para=val#sect -> archive
+            elif archive_location_type == 'url':
+
+                path = urlparse.urlparse(archive_location).path
+                archive_file = path.split('/')[-1]
+                archive_name = archive_file.split('.')[0]
+                blueprint_id = archive_name
+
+            else:
+                raise CloudifyCliError("The archive's source is not a local "
+                                       'file path or a web url')
+
+        # auto-generate blueprint id if necessary
+        if _auto_generate_ids(auto_generate_ids):
+            blueprint_id = _generate_suffixed_id(blueprint_id)
+
+        blueprints.publish_archive(archive_location,
+                                   blueprint_filename,
+                                   blueprint_id)
+    else:
+
+        if not blueprint_path:
+            blueprint_path = os.path.join(utils.get_cwd(),
+                                          DEFAULT_BLUEPRINT_PATH)
+
+        # If blueprint_id wasn't supplied, assign it to the name of
+        # folder containing the application's blueprint file.
+        if not blueprint_id:
+
+            blueprint_id = os.path.basename(
+                os.path.dirname(
+                    os.path.abspath(blueprint_path)))
+
+        # Try opening `blueprint_path`, since `blueprints.upload` expects the
+        # `blueprint_path` argument to be a file.
+        # (The reason for this is beyond me. That's just the way it is)
+
+        try:
+            with open(blueprint_path) as blueprint_file:
+
+                if _auto_generate_ids(auto_generate_ids):
+                    blueprint_id = _generate_suffixed_id(blueprint_id)
+
+                blueprints.upload(blueprint_file,
+                                  blueprint_id,
+                                  validate_blueprint)
+        except IOError as e:
+            raise CloudifyCliError("A problem was encountered while trying to "
+                                   "open the path `{0}`.\n({1})"
+                                   .format(blueprint_path, e)
+                                   )
+
+    # If deployment_id wasn't supplied, use the same name as the blueprint id.
+    if not deployment_id:
+        deployment_id = blueprint_id
+
+    # generate deployment-id suffix if necessary
+    if _auto_generate_ids(auto_generate_ids):
+        deployment_id = _generate_suffixed_id(deployment_id)
+
+    # If no inputs were supplied, and there is a file named inputs.yaml in
```

Comment:
In general, I would provide less comments and more information in the docstring. The code should be readable enough to require comments only if something is not directly clear. The docstring should supply information on the general logic so that someone reading the function knows how it's going to act.